packages/client/src/api/api.ts

```diff
@@ -13,12 +13,24 @@ import AxiosClient from "./apiClient";
 export async function signupWithEmailPassword(email: string, password: string): Promise<AxiosResponse<IAuthenticationResult>> {
   const apiUrl = `/auth/signup`;
 
-  const axiosClient = await AxiosClient.getInstance().axiosInstance.post(apiUrl, {
-    email,
-    password,
-  });
+  try {
+    const axiosClient = await AxiosClient.getInstance().axiosInstance.post(apiUrl, {
+      email,
+      password,
+    });
 
-  return axiosClient;
+    return axiosClient;
+  } catch (error) {
+    if (error instanceof AxiosError) {
+      const { response } = error;
+
+      throw {
```

Comment:
We can simply throw the error

Reply:
See there are a lot of unnecessary data which need not to be passed to the hook. We need only `response` and `status` fields

![image](https://github.com/prasenjeet-symon/intellectia/assets/105055197/2a58e75b-43db-494b-852e-9395cd3030f1)

Reply:
How we will detect the network issues? To notify the client? We need the whole AxiosError ( it is not heavy on the hook believe me )

Reply:
Yeah sure